dashboard/app/controllers/reports_controller.rb

```diff
@@ -276,34 +275,9 @@ def monthly_metrics
   def pd_progress
     authorize! :read, :reports
     script = Script.find_by(name: params[:script] || 'K5PD').cached
-    # Get all users with any activity in the script
-    users = User.joins('left join user_scripts on user_scripts.user_id = users.id').where(user_scripts: {script_id: script})
-
-    headers = nil
-    data = users.map do |user|
-      row = {}
-      row.merge!({
-                     :'ID' => user.id,
-                     :'Ops First Name' => user.ops_first_name,
-                     :'Ops Last name' => user.ops_last_name,
-                     :'Email' => user.email,
-                     :'District' => user.district.try(:name) || 'None'
-                 })
-      user_progress = summarize_user_progress(script, user)
-      percent = percent_complete(script, user)
-      script.stages.each do |stage|
-        levels = Hash[stage.script_levels.map(&:level).map do |level|
-          key = (request.format.csv? ? level.id.to_s : "<a href='#{level_path(level.id)}'>#{level.id}</a>")
-          [key, (progress = user_progress[:levels][level.id]) && progress[:status] == 'perfect' ? '1' : '0']
-        end]
-        row.merge!(levels)
-        row.merge!({:"Stage #{stage.position} Percent Complete" => percent[stage.position - 1].to_s})
-      end
-      row.merge!({:'Script Percent Complete' => percent_complete_total(script, user).to_s})
-      headers ||= row.keys
-      row.values
-    end
-    render locals: {headers: headers, data: data, script: script}
+    require 'cdo/properties'
+    locals_options = Properties.get("pd_progress_#{script.id}")
```

Comment:
I believe it is possible for us not to find the specified property; could you change the view to display an error in this case if it does not do so already?